opencti-platform/opencti-graphql/src/utils/sorting.ts

```diff
@@ -0,0 +1,32 @@
+import { getAttributeMappingFromPath, isDateNumericOrBooleanAttribute, schemaAttributesDefinition } from '../schema/schema-attributes';
+import { UnsupportedError } from '../config/errors';
+
+export const buildElasticSortingForAttributeCriteria = (orderCriteria: string, orderMode: 'asc' | 'desc') => {
+  const definition = schemaAttributesDefinition.getAttributeByName(orderCriteria);
+
+  // criteria not in schema, attempt keyword sorting as a last resort
+  if (!definition) {
+    return { [`${orderCriteria}.keyword`]: { order: orderMode, missing: '_last' } };
+  }
+
+  if (isDateNumericOrBooleanAttribute(orderCriteria)) {
+    // sorting on null dates results to an error, one way to fix it is to use missing: 0
+    // see https://github.com/elastic/elasticsearch/issues/81960
+    return { [orderCriteria]: { order: orderMode, missing: definition.type === 'date' ? 0 : '_last' } };
+  }
+
+  // for sorting by object attribute, we need the sortBy def to know which internal mapping to use
+  if (definition.type === 'object' && definition.format === 'standard') {
+    const { sortBy } = definition;
+    if (sortBy) {
+      const sortByMapping = getAttributeMappingFromPath(sortBy.path);
+      if (sortByMapping?.type === 'numeric' || sortByMapping?.type === 'boolean' || sortByMapping?.type === 'date') {
```

Comment:
This condition is not the same as the function `isDateNumericOrBooleanAttribute`?

Reply:
the functions isSomethingAttribute only check on the schema first level

Here I need to check on the inner mappings, in nested levels.

I want to do another refacto PR to change these utilities to take a SchemaAttribute as optional param, so that if not provided we take the root schema.

We could then use these utilities everywhere in all cases. But that's a potentially big refactor, these functions are everywhere. And I don't want to introduce chaos here :)